src/storage/index/IndexPolicyMaker.cpp

```diff
@@ -151,6 +157,106 @@ bool IndexPolicyMaker::exprEval(Getters &getters) {
+        default :
+            return RelationType::kNull;
+    }
+}
+
+RelationalExpression::Operator
+IndexPolicyMaker::reversalRelationalExprOP(RelationalExpression::Operator op) {
+    switch (op) {
+        case RelationalExpression::Operator::LT: {
+            return RelationalExpression::Operator::GT;
+        }
+        case RelationalExpression::Operator::LE: {
+            return RelationalExpression::Operator::GE;
+        }
+        case RelationalExpression::Operator::GT: {
+            return RelationalExpression::Operator::LT;
+        }
+        case RelationalExpression::Operator::GE: {
+            return RelationalExpression::Operator::LE;
+        }
+        default : {
+            return op;
+        }
+    }
+}
+
+bool IndexPolicyMaker::writeScanItem(const std::string& prop, const OperatorItem& item) {
+    auto op = std::get<2>(item);
+    switch (op) {
+        // for example col > 1, means the operator is GT. if col >= 1 ,means the opertor is GE.
+        // if operator is GT or GE . the 1 should a begin value.
+        case RelationalExpression::Operator::GE :
+        case RelationalExpression::Operator::GT : {
+            auto v = scanItems_.find(prop);
+            if (v == scanItems_.end()) {
+                // if the field did not exist in scanItems_, add an new one.
+                // default value is invalid VariantType.
+                scanItems_[prop] = ScanBound(Bound(toRel(op), std::get<1>(item)), Bound());
+            } else {
+                if (v->second.beginBound_.rel_ == RelationType::kNull) {
+                    // if value is invalid VariantType, reset it.
+                    v->second.beginBound_ = Bound(toRel(op), std::get<1>(item));
+                } else if (v->second.beginBound_.val_ < std::get<1>(item)) {
+                    // This might be the case where c1 > 1 and c1 > 5 , so the 5 should be save.
+                    v->second.beginBound_.val_ = std::get<1>(item);
+                }
+            }
+            break;
+        }
+        // if col < 1, means the operator is LT. if col <= 1 ,means the opertor is LE.
+        // if operator is LT or LE . the 1 should a end value.
+        case RelationalExpression::Operator::LE :
+        case RelationalExpression::Operator::LT : {
+            auto v = scanItems_.find(prop);
+            if (v == scanItems_.end()) {
+                scanItems_[prop] = ScanBound(Bound(), Bound(toRel(op), std::get<1>(item)));
+            } else {
+                if (v->second.endBound_.rel_ == RelationType::kNull) {
+                    v->second.endBound_ = Bound(toRel(op), std::get<1>(item));
+                } else if (v->second.endBound_.val_ > std::get<1>(item)) {
+                    // This might be the case where c1 < 1 and c1 < 5 , so the 1 should be save.
+                    v->second.endBound_.val_ = std::get<1>(item);
+                }
+            }
+            break;
+        }
+        case RelationalExpression::Operator::NE : {
```

Comment:
NE should be illegal based on `buildPolicy`?

Reply:
Yes, the NE should be expression.eval(), it not based on scan pair.